.travis.yml

```diff
@@ -1,53 +1,49 @@
 # ./travis.yml for MSAL android
-language: android
-jdk: oraclejdk8
-
-# Don't use the Travis Container-Based Infrastructure
-sudo: true
+dist: bionic
 
 # Build Variables
 env:
   global:
-    - BUILD_TOOLS_VERSION=28.0.3
-    - COMPILE_SDK_VERSION=28
-    - EMULATOR_API_VERSION=21
-
-    # This is to guarantee a clean gradle log
-    - TERM=dumb
-
-android:
-  components:
-    - tools
-    - platform-tools
-    - build-tools-$BUILD_TOOLS_VERSION
-    - android-$COMPILE_SDK_VERSION
-    - extra-google-m2repository
-
-    ## Emulator config ##
-    # The SDK version used for the emulator
-    - android-$EMULATOR_API_VERSION
-
-    #system images
-    - sys-img-armeabi-v7a-android-$EMULATOR_API_VERSION
+    - BUILD_TOOLS_VERSION=29.0.3
+    - API_VERSION=29
+    - ANDROID_TOOLS=commandlinetools-linux-6609375_latest.zip
+    - ARCH=x86_64
 
 before_install:
+  # download and unzip Android SDK command line tools
+  - sudo apt install openjdk-8-jdk
+  - export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64 
+  - echo $JAVA_HOME
+  #- sudo update-alternatives --set java /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java
+  - wget -nv https://dl.google.com/android/repository/$ANDROID_TOOLS
+  - unzip -q $ANDROID_TOOLS -d $HOME/sdk
+  # set SDK tools path variable and ANDROID_HOME
+  - export ANDROID_BIN=$HOME/sdk/tools/bin
+  - export ANDROID_HOME=$HOME/sdk
+  - export ANDROID_SDK_ROOT=$HOME/sdk
+  # create empty cfg file to prevent sdkmanager warning message
+  - mkdir -p $HOME/.android && touch $HOME/.android/repositories.cfg
   - chmod +x gradlew
-
-before_cache:
-  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
-  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
-cache:
-  directories:
-    - $HOME/.gradle/caches/
-    - $HOME/.gradle/wrapper/
-    - $HOME/.android/build-cache
+  - $ANDROID_BIN/sdkmanager --update --sdk_root=$ANDROID_HOME --no_https --verbose
+  - yes | $ANDROID_BIN/sdkmanager --install "tools" --sdk_root=$ANDROID_HOME --no_https --verbose | grep -v = || true
```

Comment:
Have you thought about checking in a license file?

Reply:
Yes.  I've seen that break in some builds though... so i've opted not to do that for now.  We can revisit if it's performance pain in practice